### ISSUE TITLE
Hey! Found this repository. Wanted to ask a few questions about this project 

### ISSUE THREAD

**Issue by MetalMaxMX** [ThatGuyChens/SDL-TOWER#1]
1. First of. What is the aim of this project? I understand that it is called 'SDL-TOWER' which means uncoupling from GMS quite obviously. But what else? Will it be a full accurate reconstruction of the engine with perhaps even better performance than the original game? Will it only support the original game or also the 2019 Pizza Tower demo? A full description of the project in short
2. Will there be a modding API of sorts? Perhaps some bindings to Lua/Luau along with a mention on how to program such mods
3. If 2, then how will the mods be packed and shared? 
4. How much is the completion of the project? Any estimated release dates or will it be 'comes as it goes' type?
5. Will there be any additional features outside of modding or what you can obtain from an SDL port (Custom launchers? More options to tweak video and audio? Etc.)
6. If 4's completion is at 100%. Then how is an user who acquired a copy of this game meant to interface with this port since one would have to provide their own data to actually do much of anything. Or will it be compatible with gamemaker's data.win?

I wish good luck and I personally cheer on this project! Just needed to ask some questions since it seems so promising.

**Comment by ThatGuyChens** [comment]
1. It's literally a remake of the game just for the hell of it as a hobby project, mostly basing it on the demo rn to see how much I can recreate. If it goes well I will probably follow the steam release.
2. I have been thinking about that, might do something like barotrauma and see how it goes
3. xml probably and custom dir
4. It's a hobby project so I don't know, I have school to worry about too
5. Probably Homebrew ports, I've been messing around with the SDL-3DS repo lately
6. I'll make it so the user has to dump the assets from the original game to play this for obvious reasons

**Comment by MetalMaxMX** [comment]
Thanks for answering! A few more questions 

1. As far as modding support goes. Would the idea of embedding a scripting language for modding potential be something acceptable? Even just a Lua or Wren would probably make things a bit more accessible 
2. What ways are there to contribute to this project as far as testing/localization goes? Or too early to tell? (There are a few projects on Github wanting to localize the game so perhaps localization support is important)

**Comment by ThatGuyChens** [comment]
1. I'm still thinking about it, maybe Lua?
2. As far as Localization goes, I really don't know. Might ask friends for help but will see.

**Comment by MetalMaxMX** [comment]
All right, once again, thanks for answering. 
Huh, do tell if there's going to be a website or a means of communication for how this project is going to be handled (for the rest of the people). Though I imagine that's out of scope given the early state of the project

**Comment by MetalMaxMX** [comment]
Right, forgot to ask. But what language is this port going to be implemented with? I've seen that you started with C++ files but I'm not sure if that's going to be the only primary language or if it will be a combination of C and C++?

**Comment by ThatGuyChens** [comment]
It's gonna be C++ but might incorporate other languages, rn having to move everything to make cause having issues with Cmake

**Comment by MetalMaxMX** [comment]
Good job on getting it to work with make!

Huh, another question. How many dependencies are you going to bring in this project? I understand that you'll use SDL (presumably SDL2) but what else? Perhaps open_al for audio? Something else?

**Comment by ThatGuyChens** [comment]
> All right, once again, thanks for answering. Huh, do tell if there's going to be a website or a means of communication for how this project is going to be handled (for the rest of the people). Though I imagine that's out of scope given the early state of the project

1. I'm thinking of opening my discord I made for a Source Engine fork but still kinda too early tbh
2. As for dependencies, I need to check what works on multiple platforms since as I said I want it to also work as a homebrew game for anyone who has the time to compile it for said platform they might be using

I can't find much time to work on this for the time being due to having to search for a new laptop to buy for school and because I need to repair a laptop for my parents (just bought replacement parts)

**Comment by MetalMaxMX** [comment]
Oh, well, take your time then. 

If I can make a suggestion surrounding dependencies. This engine (https://github.com/nCine) might be worth studying. A reimplementation of another platformer (Jazz Jackrabbit 2) has been done on it with great success and the engine is also built on SDL2 to boot. It has quite a breadth of supported platforms and the dependencies are generic enough that you could use them on homebrew ports (check https://github.com/nCine/nCine under the dependencies and platforms section)

Another possibility is to build and study from the HatchGameEngine https://github.com/Lactozilla/HatchGameEngine since that also supports homebrew ports like the Nintendo Switch and has a very Retro Engine-like quality (RSDK I mean) 

Whatever path you choose from, again, take your time with this and focus on what you need to do.

**Comment by MetalMaxMX** [comment]
Oh, if you have the time to write again. I'd like to know if you have plans for the way the game handles FPS. As you may know, GM does not implements deltatime (rather, it has an implementation but is rarely used) so it ties the physics to the framerate, as a result, a slower device would have slower gameplay and conversely, removing the FPS cap just makes the game unplayable on faster devices. This was a relatively common complaint in some of the questions asked in the Pizza Tower forums at Steam, wonder if this could have a solution
